Process macro for deriving `Into<Value>` trait implementations for enums and structs.

chrsoo opened the pull request:
Discussed in #1. this PR adds support for...

* Using a `#[handybars_value]` macro to automatically generate `Into<Value>` trait implementations, i.e. to automatically treat enums and structs as Handybars `Value` variants.
* Automatically converting rust number primitives into `Value::String` variants.
* Building for Mac, Linux, and Windows on Github using GitHub actions.

### Notes

* The `handybars_value`macro is implemented as a separate `handybars_attribute` crate located in the `attribute` sub directory.
* A workspace is added in `Cargo.toml` of the parent crate; lint, build and test with the following commands:
```
cargo clippy --workspace
cargo build --workspace
cargo test --workspace
```
* The `handybars_attribute` version is set to be the same as the parent crate.
* The GitHub build will fail if Clippy is not satisfied... :-)

### Tips
You might already be up to speed, but the following proved invaluable when building macros using Visual Studio Code with Rust macros:
* Rust Analyser does not catch up on code changes in the attribute crate. To mitigate this, the window can be refreshed (Mac: Cmd-Shift-A, run "Developer: Reload Window").
* Macros can be expanded to see the generated code (Mac: Cmd-Shift-A, run "rust analyser: Expand macro recursively at caret")

chrsoo commented:
And yes, regarding feature gates, I am not sure how that would apply as current limitations in Rust requires macros to be implemented as a separate crate.

Given that this attribute macro crate must depend on the crate for which it generates code, we cannot include a feature gate in the main project crate, this would create a circular dependency.

I could think of ways of working around this, but it adds complexity.

In any case, the main requirement of not pulling in the `syn` and `quote` dependencies is met as this only happens for anybody using the attribute.

0x00002a commented:
Firstly we should be exporting the proc macro from the root crate with `pub use handybars_attribute::handybars_value` as really the fact it's a separate crate is an implementation detail that is due to the way cargo works. This is how other crates do it as well. Doing it this way is why we have a feature gate, add `handybars_attribute` as an optional dependency behind a feature gate.

The dep cycle comes about because handybars is used in the tests, you can put handybars in the dev dependencies only and it'll work fine.

I think `handybars_attribute` should be called something like `handybars_macros`, this is the unofficial convention for these things

chrsoo commented:
> The dep cycle comes about because handybars is used in the tests, you can put handybars in the dev dependencies only and it'll work fine.
Yes, will do. Obvious solution, should have looked around better.

> I think handybars_attribute should be called something like handybars_macros, this is the unofficial convention for these things
I followed what I think was the official naming convention, but now I can't find the reference... Anyhow I was also thinking that we might want to call the crate `handybars_macros` so I will rename it.

chrsoo commented:
Done.

* Renamed the `handybars_attribute` crate to `handbars_macros`.
* Added an optional dependency to `handybars_macros` in the parent crate.
* Removed the dependency to the parent crate in `handybars_macro`.
* Introduced the optional `macros` feature.
* Republished the `handybars_value` macro in `lib.rs`.
* Moved `tests/handybars_macros.rs` to the parent crate.
* Moved the macro usage examples to the README.md in the parent crate.

Struggled a bit with the feature gate and the documentation examples in the main `README.md`. I did not manage to find out how to specify that that the new `macros` feature should be enabled for all example code and ended up repeating a `cfg` macro for every item in the examples. It comes out quite ugly on GitHub:

>  ```rust
> # #[cfg(feature = "macros")]
> # use handybars::handybars_value;
> # #[cfg(feature = "macros")]
> #[handybars_value]
> enum SimpleEnumProp {
>    A,
>    B,
> }

The rendered rust doc looks better though:
![image](https://github.com/user-attachments/assets/d5f13c6e-955b-495e-a2ef-ddd016d5d3cf)

If you know how to stay DRY and not repeat the line
```rust
> # #[cfg(feature = "macros")]
```
... for every item in the examples, it would be great!

chrsoo commented:
> * Building for Mac, Linux, and Windows on Github using GitHub actions.

I just realised I thought I added a Clippy support and multiple targets by accepting the suggested GitHub build file, but in reality it was already there...  So please forget anything I said about Clippy and automated builds!

Very embarrassing, too fast sometimes.

0x00002a commented:
Amazing. Few small spelling issues but I'll fix those up myself, thanks!
